packaging/scripts/test-telemetry-agent.sh

```diff
@@ -0,0 +1,232 @@
+                echo "Percona telemetry agent is installed. Removing..."
+                apt-get remove -y percona-telemetry-agent
+                echo "Removing Percona repository files..."
+                rm -f /etc/apt/sources.list.d/percona-*.list
+                apt-get update
+            else
+                echo "Percona telemetry agent is not installed."
+            fi
+            ;;
+        *)
+            echo "Unsupported OS: ${OS}"
+            exit 1
+            ;;
+    esac
+}
+
+install_percona_release() {
+  case "$OS" in
+          ol)
+              # Oracle Linux
+              if [ "$VERSION_ID" == "8" ] || [ "$VERSION_ID" == "9" ]; then
+                  yum install -y https://repo.percona.com/yum/percona-release-latest.noarch.rpm
+              else
+                  echo "Unsupported Oracle Linux version: ${VERSION_ID}"
+                  exit 1
+              fi
+              ;;
+          amzn)
+            # Amazon Linux
+            if [ "$VERSION_ID" == "2" ] || [ "$VERSION_ID" == "2023" ]; then
+              yum install -y https://repo.percona.com/yum/percona-release-latest.noarch.rpm
+            else
+              echo "Unsupported Amazon Linux version: ${VERSION_ID}"
+              exit 1
+            fi
+            ;;
+          debian | ubuntu)
+              if [ "$VERSION_ID" == "11" ] || [ "$VERSION_ID" == "12" ] || [ "$VERSION_ID" == "20" ] || [ "$VERSION_ID" == "22" ] || [ "$VERSION_ID" == "24" ]; then
+                  apt-get update
+                  apt-get install -y wget gnupg2 lsb-release curl systemd
+                  wget https://repo.percona.com/apt/percona-release_latest.$(lsb_release -sc)_all.deb
+                  dpkg -i percona-release_latest.$(lsb_release -sc)_all.deb
+              else
+                  echo "Unsupported Debian/Ubuntu version: ${VERSION_ID}"
+                  exit 1
+              fi
+              ;;
+          *)
+              echo "Unsupported OS"
+              exit 1
+              ;;
+      esac
+}
+
+check_telemetry_agent_logs() {
+  if [ ! -d /var/log/percona/telemetry-agent ]; then
+      echo "telemetry-agent log location is missing"
+      exit 1
+  fi
+
+  if [ ! -f /var/log/percona/telemetry-agent/telemetry-agent.log ]; then
+      echo "telemetry-agent log file is missing"
+      exit 1
+  fi
+
+  if [ ! -f /var/log/percona/telemetry-agent/telemetry-agent-error.log ]; then
+      echo "telemetry-agent error log file is missing"
+      exit 1
+  fi
+
+  echo "validated telemetry-agent log files"
+}
+
+check_percona_telemetry_version() {
+  output=$(percona-telemetry-agent --version)
+
+  version=$(echo "$output" | grep "Version:" | awk '{print $2}')
+  commit=$(echo "$output" | grep "Commit:" | awk '{print $2}')
+  build_date=$(echo "$output" | grep "Build date:" | awk '{print $3}')
+
+  if [ -z "$version" ]; then
+      echo "Error: Version information is empty"
+      exit 1
+  fi
```

Comment:
@idoqo Could we also validate if the version printed is correct?

Reply:
@idoqo We can either update the script for each release or the better way would be to add a parameter named  something like "TA_version" to the test_telemetry_agent jenkins job and comparing this value with the version output from the percona-telemetry-agent binary.